shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -921,6 +931,36 @@ - (void)waitForFirstFrame:(NSTimeInterval)timeout
   });
 }
 
+- (FlutterEngine*)spawnWithEntrypoint:(/*nullable*/ NSString*)entrypoint
+                           libraryURI:(/*nullable*/ NSString*)libraryURI {
+  NSAssert(_shell, @"Spawning from an engine without a shell (possibly not run).");
+  FlutterEngine* result = [[FlutterEngine alloc] initWithName:_labelPrefix
+                                                      project:_dartProject.get()
+                                       allowHeadlessExecution:_allowHeadlessExecution];
+
+  flutter::Settings settings = _shell->GetSettings();
```

Comment:
Can we extract parts of this into a private method to share between here and createShell? To minimize the risk of someone updating one and not the other as much as possible?

Reply:
Yep, I did that in `setupShell:withObservatoryPublication:` and `SetEntryPoint`.  There is very little happening here, mostly boilerplate around lambdas.